### PR TITLE
Update project structure

### PR DESCRIPTION
**Pull request by kairen**
do-not-merge/work-in-progress

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,28 +1,35 @@
 # OpenStack-Exporter
+Simple Prometheus exporter for OpenStack.
 
-Simple Prometheus exporter for OpenStack
+## Build
+First clone the repo, and excute `Makefile` using make tool:
```
Comment (iGene) [inline README.md]: typo here, should be execute

Comment (ching-kuo) [inline README.md]: Typo, should be execute

File: cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		}
+	}
+}
+
+func loadOpenStackConfig(path string) {
+	viper.SetConfigFile(path)
+	err := viper.ReadInConfig()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Fatal error config file: %s \n", err)
+		os.Exit(1)
```
Comment (ching-kuo) [inline cmd/main.go]: Could change to log.Fatalf for more consistency on logging.

File: README.md [hunk 3]
```diff
@@ -1,28 +1,35 @@
...
+$ make build_image
+```
 
-Copy the sample configuraion file and fill in
+## Deployment
+Copy the sample configuraion file and fill in:
 
 - OpenStack Username
 - OpenStack Password
 - OpenStack Keystone Endpoint
```
Comment (ching-kuo) [inline README.md]: Should change this part into 3 sections, each section describing how to configure it using the below methods.

1. Configuration file
2. Command line options
3. Environment variables

Also there's a typo (configraion -> configration)

**Review by ching-kuo — CHANGES_REQUESTED** [review]
RegionName should be pass to each collector to create its client, but this can be change later on.

There are other minor doc problems that should be addressed

File: README.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 Simple Prometheus exporter for OpenStack.
 
 ## Build
-First clone the repo, and excute `Makefile` using make tool:
+First clone the repo, and should be execute `Makefile` using make tool:
```
Comment (ching-kuo) [inline README.md]: should be execute -> execute

File: cmd/main.go [hunk 5]
```diff
@@ -33,8 +33,8 @@ func init() {
 	pflag.StringVar(&configPath, "config", "", "Load the OpenStack config from path.")
 	pflag.StringVar(&endpoint, "keystone-url", os.Getenv("OS_AUTH_URL"), "URL for the OpenStack Keystone API.")
 	pflag.StringVar(&domain, "domain-name", os.Getenv("OS_DOMAIN_NAME"), "Domain name for the OpenStack Keystone.")
-	pflag.StringVar(&password, "password", os.Getenv("OS_PASSWORD"), "Password for the OpenStack Keystone.")
 	pflag.StringVar(&user, "username", os.Getenv("OS_USERNAME"), "User for the OpenStack Keystone.")
+	pflag.StringVar(&password, "password", os.Getenv("OS_PASSWORD"), "Password for the OpenStack Keystone.")
 	pflag.StringVar(&project, "project-name", os.Getenv("OS_PROJECT_NAME"), "Project Name for the OpenStack Keystone.")
 	pflag.StringVar(&region, "region-name", os.Getenv("OS_REGION_NAME"), "Region Name for the OpenStack Keystone.")
```
Comment (ching-kuo) [inline cmd/main.go]: for the OpenStack Keystone -> for OpenStack Keystone

File: README.md [hunk 6]
```diff
@@ -15,18 +15,40 @@ $ make build_image
 ```
 
 ## Deployment
-Copy the sample configuraion file and fill in:
+Should change this part into 3 sections to selecting each section describing how to configure it using the below methods.
```
Comment (ching-kuo) [inline README.md]: Change it to
"Configuring OpenStack exporter can be done in 3 ways, choose the one that is best suitable for your environment."

**Review by ching-kuo — CHANGES_REQUESTED** [review]
Besides some minor problems, coding part LGTM

**Review by ching-kuo — APPROVED** [review]

**Comment by ching-kuo** [comment]
LGTM

**Pull request merged**